### ISSUE TITLE
Fixed bug in PMD seemingly still present in eclipse-pmd

### ISSUE THREAD

**Issue by Frettman** [eclipse-pmd/eclipse-pmd#34]
I'm using the latest version 3.3.0 of eclipse-pmd. I have a ruleset that includes `UselessOverridingMethod`.

PMD 6.22.0 fixed a bug in that rule, so the following example would no longer raise a violation:

```java
class TestA
{
  protected void hello()
  {
    System.out.println("Hello, world!");
  }
}

class TestB extends TestA
{
  @Override
  public void hello()
  { // Elevate the access level - PMD should no longer report this as useless
    super.hello();
  }
}
```
That version of PMD should be included in eclipse-pmd since 1.27. And yet, the latest version of eclipse-pmd still shows a violation for that example.
I have compared this with the output of the Maven plug-in (using the exact same PMD version as eclipse-pmd), but there I correctly don't see that violation.
Do you have any idea what could cause this?

**Comment by acanda** [comment]
The rule `UselessOverridingMethod` uses type resolution to check for the elevated access level. Unfortunately, eclipse-pmd does not support type resolution.

**Comment by Frettman** [comment]
Ah, unfortunate, but that explains it. Thank you.

**Comment by eclipse-pmd** [comment]
I just released eclipse-pmd 3.4 with support for type resolution for Java rules. It should be able to resolve types as long as they are in your current Eclipse project, in one of its libraries or in a referenced Java project.

**Comment by Frettman** [comment]
I installed 3.4, but the violation is still there using the example from above.

**Comment by eclipse-pmd** [comment]
Do you use the exact example from above, in a Java project, with PMD enabled? When I copy the code snippet from above, paste it into a Java file and enable PMD, then I get a warning for the System.out.println and one for having two classes in one file, but no warning for UselessOverridingMethod.

Can you check that you have eclipse-pmd 3.4 installed, both features "eclipse-pmd Core" and "eclipse-pmd Java Support"?
Help > About Eclipse IDE > Installation Details 
![grafik](https://user-images.githubusercontent.com/62813972/217373954-5b35a9b3-c92a-4c01-baf0-21087bc8b97d.png)

**Comment by Frettman** [comment]
I had already verified the correct plug-in version and the new plug-in. I had refreshed/cleaned/updated the project. And I had watched the PMD problem markers disappear when disabling PMD for the project and reappear when re-enabling it, including UselessOverridingMethod. Eclipse was even offering me to add the `@SuppressWarnings("PMD.UselessOverridingMethod")`.

But just to be extra sure, I have just now created a new Java project and moved over those two files. At first they had the same problem marker for `UselessOverridingMethod`, even though I hadn't even enabled PMD yet for the project! Maybe Eclipse just moved them with the files? Anyway, after enablinig PMD for the project, that marker was gone and the one for System.out.println was still there. So everything looked good. Then I re-tested it in the previous project and it worked too...
So...it works unless Eclipse has one of its bad days...?
Anyway, I'll permanently enable that check now and I'll report back if the problem reappears and I can maybe add some more useful information. Otherwise, thank you for taking the time to implement this.
